### PR TITLE
Initial working version

### PR DESCRIPTION
**Pull request by makermelissa**
This is an initial working version. Most function were taken from the MagTag library in order to get the latest versions of things. It is not meant to be used by itself, so there are no examples.

### CODE REVIEW

**Comment by makermelissa** [comment]
To test, you can use https://github.com/makermelissa/Adafruit_CircuitPython_MagTag/tree/refactor which should run any existing MagTag scripts without changes. I tested on MagTag the examples already.

**Review by FoamyGuy — APPROVED** [review]
I tested this successfully with the linked MagTag PR branch and the progress display scripts that I've been working on.

**Comment by makermelissa** [comment]
Thanks for testing

File: adafruit_portalbase/graphics.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+                y=position[1],
+            )
+        else:
+            raise RuntimeError("Unknown type of background")
+        self._bg_group.append(self._bg_sprite)
+        gc.collect()
+
+    def qrcode(
+        self, qr_data, *, qr_size=1, x=0, y=0, qr_color=0x000000
+    ):  # pylint: disable=invalid-name
```
Comment (tannewt) [inline adafruit_portalbase/graphics.py:110]: I'd love to see this split out into it's own widget like label. That way we can use it outside of portal.

Comment (makermelissa) [reply adafruit_portalbase/graphics.py:110]: 🤔  Please elaborate. I think I know what you mean, but want to be sure. Do you mean as part of an existing library or a maybe new library?

Comment (tannewt) [reply adafruit_portalbase/graphics.py:110]: I was thinking a new library or even part of the QRCode library. It'd be an object that represents the QRCode on screen. The constructor would take the parameters here and the object would inherit from Group so you could use and manipulate it like a normal group.

**Pull request merged**